content/My favorite extensions for programming Ruby in Visual Studio Code.md

```diff
@@ -0,0 +1,106 @@
+- On-demand gutter blame annotations, including a heatmap, for the whole file
+- On-demand gutter heatmap annotations to show how recently lines were changed, relative to all the other changes in the file and to now (hot vs. cold)
+- On-demand recent changes annotations to highlight lines changed by the most recent commit
+- An much [more](https://gitlens.amod.io/#features).
+
+![Gitlens](images/gitlens-preview.gif)
+
+## [Ruby](https://marketplace.visualstudio.com/items?itemName=rebornix.Ruby)
+
+This extension provides enhanced Ruby language and debugging support, one of the coolest things about this extension is the syntax support to our ruby files, is a **must-have** if you are a Ruby developer.
+
+**Features:**
+
+- Automatic Ruby environment detection with support for rvm, rbenv, chruby, and asdf
+- Lint support via RuboCop, Standard, and Reek
+- Format support via RuboCop, Standard, Rufo, and RubyFMT
+- Semantic code folding support
+- Semantic highlighting support
+- Basic Intellisense support
+
+![VSCode Ruby](images/vsrubycode.png)
+
+## [Ruby Solargraph](https://marketplace.visualstudio.com/items?itemName=castwide.solargraph)
+
+Another **must-have** extension if you are a Ruby developer, gives Visual Studio Code some features that can be found in IDE's like "Go to definition" for classes, modules, and methods. Also provides documentation and code completion. If you are learning Ruby, this extension will help you a lot.
+
+**Features:**
+
+- Context-aware suggestions and documentation for the Ruby core
+- Detection of some variable types and method return values (e.g., String.new. returns String instance methods)
+- Identification of local, class, and instance variables within the current scope
+- Find references and rename symbols (experimental as of solargraph gem v0.22.0)
+- Support for gems
+- Near-complete support for the Ruby core and stdlib
+- Partial support for Ruby on Rails
+
+![Ruby Solargraph](images/rubysolargraph.gif)
+
+## [ruby-rubocop](https://marketplace.visualstudio.com/items?itemName=misogi.ruby-rubocop)
+
+Having rubocop configured in our ruby project will help us to maintain the quality of the code, especially if our project is being developed by a team. This extension will let us autoformat and correct our code basing it the directives written in the .rubucop.yml file. With only a key shortcut your code will be formatted in a second, is fast, is good, is a **must-have** extension.
+
+**Features:**
+
+- Lint by executing the command "Ruby: lint by rubocop" (cmd+shift+p and type command)
+- Auto invoke when saving file
+- Auto correct command "Ruby: autocorrect by rubocop"
+
+![Ruby Solargraph](images/rubyrubocop.gif)
+
+## [Prettier](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode)
+
+This extension will help us to format the code of Javascript, HTML, CSS, JSON, and many more files. Is simple and quick, but is not so much configurable like other code-formatter extensions.
```

Comment:
```suggestion
This extension will help you to format the code of Javascript, HTML, CSS, JSON, and many more files. Is simple and quick, but is not so much configurable like other code-formatter extensions.
```

same comment here

Reply:
I would say `is not as configurable as others...`